[ref #44] アカウント登録時に個人情報保持についての同意を得る(被験者)

radcats opened the pull request:
(修正点)
participants_controller.rb 同意分表示用のアクションを追加
participant.rb validationにcheckboxにチェックが入っているかの判定を追加
_form.html.erb typeがallの場合checkboxと同意分へのリンクを表示するように修正
routes.rb 同意分のページへのパスを追加
participants_controller_test.rb agreement_pathで同意分のページを開けるかのテストを追加
participant_test.rb agreementがfalseの場合をはじけるかのテストを追加
agreement_html.erb 同意分ページのview

radcats committed:
[ref #44] アカウント登録時に個人情報保持についての同意を得る(被験者)

(修正点)
participants_controller.rb 同意分表示用のアクションを追加
participant.rb validationにcheckboxにチェックが入っているかの判定を追加
_form.html.erb typeがallの場合checkboxと同意分へのリンクを表示するように修正
routes.rb 同意分のページへのパスを追加
participants_controller_test.rb agreement_pathで同意分のページを開けるかのテストを追加
participant_test.rb agreementがfalseの場合をはじけるかのテストを追加
agreement_html.erb 同意分ページのview

okabi commented on config/routes.rb:
```diff
@@ -74,4 +74,7 @@
 
+  # 個人情報取り扱い説明ページ
+  get   '/agreement',  to: 'participants#agreement'
+
```
個別にページ作るまでもない(スクロールボックスとかに表示するで良い)気がします。

okabi commented on test/models/participant_test.rb:
```diff
@@ -179,4 +179,9 @@ def setup
+  test "agreement should not be false" do
+    @participant.agreement = false
+    assert_not @participant.valid?
+  end
```
ちゃんとテスト書けてて Good！ 😄

okabi reviewed (commented):
UI 考えるとスクロールボックスの方が良い気がしますが、
そこは判断お願いしますー

radcats committed:
[ref #44] アカウント登録時に個人情報保持についての同意を得る(被験者)

(修正点)
participants_controller.rb agreementアクションを削除
_form.hrml.erb 同意文をスクロールボックスで表示するように変更
routes.rb agreementパスを削除
participants_controller_test.rb agreementパスのテストを削除
_agreement.heml.erb 部分テンプレートに修正

radcats committed:
[ref #44] アカウント登録時に個人情報保持についての同意を得る(被験者)

(修正点)
participant.rb 同意文格納用の引数追加
_form.html.erb  form_text_areaを利用しレイアウトを揃えた
add_agreement_to_participants.rb  agreement引数追加用のmigrationファイル

yegawa commented on db/migrate/20170407142910_add_agreetext_to_participants.rb:
```diff
@@ -0,0 +1,5 @@
+class AddAgreetextToParticipants < ActiveRecord::Migration[5.0]
+  def change
+    add_column :participants, :agreetext, :string
```
ただの垢登録の同意文だから一意で良いと思うし
Participantモデルにagreetextを含める必要はないと思います。

yegawa commented on app/views/participants/_form.html.erb:
```diff
@@ -20,10 +20,8 @@
-      <%= render :partial => "agreement" %>
-    </textarea>
+    <% participant.agreetext = File.open("#{Rails.root}/app/views/participants/_agreement.html.erb","rb").read.force_encoding("UTF-8") %>
+    <%= form_text_area f, :agreetext, '個人情報の取り扱い' %> 
```
`text_area_tag` を使えば対象のモデルがなくてもテキストエリアを設置できます。
レイアウトを自動でやるには `app/helpers/application_helper.rb` に `form_xxxx` に似せた感じで
`form_xxxx_tag` を書いてやる必要が出てきます。
`form_xxxx_tag` については用途を考えると `value:` （デフォ値）を指定できるようにすると良いと思います。

yegawa commented on app/views/participants/_form.html.erb:
```diff
@@ -20,10 +20,8 @@
-    <textarea cols="100" '個人情報の取り扱い' rows="6" style="overflow:scroll;vertical-align:middle" readonly=true>
-      <%= render :partial => "agreement" %>
-    </textarea>
+    <% participant.agreetext = File.open("#{Rails.root}/app/views/participants/_agreement.html.erb","rb").read.force_encoding("UTF-8") %>
```
ここちょっと審議。
個人的にはviewで仰々しく読み込むことに違和感がある。
ここで同じようなことやってるけど、まあこっちもコントローラで無理やり取ってきてるし変っちゃ変かも…。
http://stackoverflow.com/questions/35639091/how-do-i-render-a-partial-in-a-text-area-in-a-rails-form

yegawa reviewed (changes requested):
ファイル読み込むところはちょっと分からんけど、それ以外のところは修正したほうが良い気がします。
form_xxxx_tagの実装はコード行数が多くて少し重くなってしまうかもしれないです、ごめんなさい…

radcats committed:
[ref #44] アカウント登録時に個人情報保持についての同意を得る(被験者)

(修正点)
application_helper.rb  form_tagで使う関数の定義 (未 select,collection_select)
participant.rb agreetext引き数削除
_form.html.erb 同意文表示にtext_area_tagを使用するように変更

radcats committed:
[ref #44] アカウント登録時に個人情報保持についての同意を得る(被験者)

(修正点)
application_helper.rb form_tag用のselect,collection_selectの追加

okabi commented on app/views/participants/_agreement.html.erb:
```diff
@@ -0,0 +1,10 @@
+
```
スクロールボックスの表示的に最初に改行が入ってると微妙なので、消しといて欲しいです。

okabi reviewed (approved):
同意文の最初の改行を消してもらったらマージしても良さそう。
ただ、少し気になる点もあります。

- 使用されていない `xx_tag` が量産されている
- `File.open` のところが少し仰々しい気もするけど、代替案も出せないので難しいですね…。

radcats committed:
[ref #44] アカウント登録時に個人情報保持についての同意を得る(被験者)

(修正点)
application_helper.rb 使ってないform_tag関数の削除
agreement.html.erb 改行の削除

okabi reviewed (approved):
良さそう 👍
